### ISSUE TITLE
TestMultiRangeEmptyAfterTruncate leaks gossip bootstrap worker

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3362]
The following test appears to have failed:

[#9958](https://circleci.com/gh/cockroachdb/cockroach/9958):

```
W1208 22:28:20.740094 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
W1208 22:28:21.001926 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
I1208 22:28:21.002050 596 client/db.go:483  failed batch: failed to send RPC request Server.Batch: client is unhealthy
W1208 22:28:21.008323 596 rpc/client.go:249  unexpected EOF
I1208 22:28:21.008493 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestKVDBTransaction (0.39s)
	db_test.go:213: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestAuthentication
I1208 22:28:21.052676 596 rpc/clock_offset.go:154  monitoring cluster offset
I1208 22:28:21.062721 596 multiraft/multiraft.go:579  node 1 starting
I1208 22:28:21.063294 596 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:28:21.063498 596 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:28:21.063640 596 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1208 22:28:21.063735 596 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:28:21.063825 596 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:28:21.063902 596 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6
--
W1208 22:28:21.134254 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
W1208 22:28:21.354774 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
I1208 22:28:21.354907 596 client/db.go:483  failed batch: failed to send RPC request Server.Batch: client is unhealthy
W1208 22:28:21.374996 596 rpc/client.go:249  unexpected EOF
I1208 22:28:21.375183 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestAuthentication (0.37s)
	db_test.go:258: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestRangeLookupWithOpenTransaction
I1208 22:28:21.411283 596 rpc/clock_offset.go:154  monitoring cluster offset
I1208 22:28:21.413596 596 multiraft/multiraft.go:579  node 1 starting
I1208 22:28:21.414267 596 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:28:21.414528 596 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:28:21.414665 596 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1208 22:28:21.414754 596 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:28:21.414842 596 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:28:21.414917 596 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6
--
I1208 22:28:21.781387 596 server/node.go:251  Started node with [[]=] engine(s) and attributes []
I1208 22:28:21.781960 596 server/server.go:219  starting https server at 127.0.0.1:47517
W1208 22:28:21.795814 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
W1208 22:28:22.064818 596 client/rpc_sender.go:93  failed to send RPC request Server.Batch: client is unhealthy
I1208 22:28:22.064959 596 client/db.go:483  failed batch: failed to send RPC request Server.Batch: client is unhealthy
--- FAIL: TestMultiRangeBatchBounded (0.33s)
	dist_sender_server_test.go:98: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestMultiRangeEmptyAfterTruncate
I1208 22:28:22.108441 596 rpc/clock_offset.go:154  monitoring cluster offset
I1208 22:28:22.134553 596 multiraft/multiraft.go:579  node 1 starting
I1208 22:28:22.135268 596 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:28:22.135413 596 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:28:22.135534 596 multiraft/multiraft.go:999  node 1 campaigning because initial confstate is [1]
I1208 22:28:22.135606 596 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:28:22.135681 596 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:28:22.135747 596 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6
--
tba4548c6  127.0.0.1:0  22:28:22.714332  0             ·coordinator stops                            kv/txn_coord_sender.go:496

I1208 22:28:22.719944 596 stopper.go:241  draining; tasks left:
1      storage/replica.go:1441
I1208 22:28:22.729629 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestMultiRangeEmptyAfterTruncate (0.81s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:524 +0x361
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
Name                  Origin       Ts               Dur         Desc                     File
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:23.393426  8.130873ms  sending batch            kv/txn_coord_sender.go:309
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:23.393497  26.335µs    ·meta descriptor lookup  kv/dist_sender.go:519

I1208 22:28:23.405869 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestMultiRangeScanReverseScanDeleteResolve (0.67s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
RdRgRv@1449613703.989899874,1  127.0.0.1:0  22:28:24.014052  251.667µs  ·executing 1 requests  storage/store.go:1281
RdRgRv@1449613703.989899874,1  127.0.0.1:0  22:28:24.014084  207.115µs  ··read-only path       storage/replica.go:579
RdRgRv@1449613703.989899874,1  127.0.0.1:0  22:28:24.014098  2.566µs    ···command queue       storage/replica.go:754

I1208 22:28:24.017347 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestMultiRangeScanReverseScanInconsistent (0.63s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:24.666668  14.36µs     ·meta descriptor lookup       kv/dist_sender.go:519
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:24.666701  1.893906ms  ·sending RPC                  kv/dist_sender.go:414
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:24.666818  0           ··sending to 127.0.0.1:52107  rpc/send.go:155

I1208 22:28:24.670784 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestSingleRangeReverseScan (0.62s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
Name                  Origin       Ts               Dur          Desc                     File
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:25.405296  12.673202ms  sending batch            kv/txn_coord_sender.go:309
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:25.405328  4.679398ms   ·meta descriptor lookup  kv/dist_sender.go:519

I1208 22:28:25.423357 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestMultiRangeReverseScan (0.77s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
Name                  Origin       Ts               Dur         Desc                     File
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:26.442413  22.23771ms  sending batch            kv/txn_coord_sender.go:309
RdRgRv@0.000000000,0  127.0.0.1:0  22:28:26.442451  21.675µs    ·meta descriptor lookup  kv/dist_sender.go:519

I1208 22:28:26.468487 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestReverseScanWithSplitAndMerge (1.05s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
WrRg@0.000000000,0  127.0.0.1:0  22:28:26.930241  3.002793ms  ·sending RPC                                                                         kv/dist_sender.go:414
WrRg@0.000000000,0  127.0.0.1:0  22:28:26.930448  0           ··sending to 127.0.0.1:38351                                                         rpc/send.go:155
WrRg@0.000000000,0  127.0.0.1:0  22:28:26.933689  0           ·error: storage/store.go:116: start key in [/Min,"z") must be greater than LocalMax  kv/txn_coord_sender.go:405

I1208 22:28:26.938800 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestBadRequest (0.47s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
td3c9b346  127.0.0.1:0  22:28:27.662704  0            ·coordinator stops                            kv/txn_coord_sender.go:496

I1208 22:28:27.669664 596 stopper.go:241  draining; tasks left:
1      storage/replica.go:1441
I1208 22:28:27.677728 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestNoSequenceCachePutOnRangeMismatchError (0.75s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
ta36ee9a0  127.0.0.1:0  22:28:28.323021  20.112517ms  ·sending RPC                  kv/dist_sender.go:414
ta36ee9a0  127.0.0.1:0  22:28:28.323148  0            ··sending to 127.0.0.1:41052  rpc/send.go:155
ta36ee9a0  127.0.0.1:0  22:28:28.343447  0            ·coordinator stops            kv/txn_coord_sender.go:496

I1208 22:28:28.353629 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestPropagateTxnOnError (0.68s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
t603a3b40  127.0.0.1:0  22:28:29.017037  0           ·coordinator stops            kv/txn_coord_sender.go:496

I1208 22:28:29.021619 596 stopper.go:241  draining; tasks left:
1      storage/replica.go:1441
I1208 22:28:29.027438 596 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- FAIL: TestPropagateTxnOnPushError (0.67s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:517 +0x3ca
		github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82080c3c0, 0xc82082e900)
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
		
		github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
			/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:548 +0x2bf
--
			/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
		github.com/cockroachdb/cockroach/kv_test.TestMain(0xc820641f10)
			/go/src/github.com/cockroachdb/cockroach/kv/main_test.go:35 +0x2f
		main.main()
			github.com/cockroachdb/cockroach/kv/_test/_testmain.go:206 +0x20a
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	38.234s
=== RUN   TestHeartbeatSingleGroup
I1208 22:27:45.478162 571 multiraft/multiraft.go:579  node 1 starting
I1208 22:27:45.478814 571 multiraft/multiraft.go:579  node 2 starting
I1208 22:27:45.479727 571 raft/raft.go:441  [group 1] 1 became follower at term 5
I1208 22:27:45.480031 571 raft/raft.go:234  [group 1] newRaft 1 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:27:45.480576 571 raft/raft.go:441  [group 1] 2 became follower at term 5
I1208 22:27:45.480768 571 raft/raft.go:234  [group 1] newRaft 2 [peers: [1,2], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1208 22:27:45.481098 571 raft/raft.go:521  [group 1] 1 is starting a new election at term 5
I1208 22:27:45.481365 571 raft/raft.go:454  [group 1] 1 became candidate at term 6
I1208 22:27:45.481558 571 raft/raft.go:503  [group 1] 1 received vote from 1 at term 6

```
Please assign, take a look and update the issue accordingly.

**Comment by mrtracy** [comment]
Another instance: [#9971](https://circleci.com/gh/cockroachdb/cockroach/9971)

**Comment by mrtracy** [comment]
I'm basically unable to get a clean CircleCI build anymore, I've had three builds in a row fail with these same failures, including one build without cache.

**Comment by tschottdorf** [comment]
wanna send a PR disabling the test? Seems like The Right Thing To Do(TM) at this point.

**Comment by mrtracy** [comment]
Well it's actually like ten tests, but I just checked my latest CircleCI and it was a different flake, which actually doesn't look like a flake.

**Comment by petermattis** [comment]
My understanding is that the failure is really in `TestKVDBTransaction` and will be fixed by #3428.

**Comment by tschottdorf** [comment]
apparently the case.
